pkg/api/controller.go

```diff
@@ -802,27 +810,71 @@ func (c *Controller) ListUsers(w http.ResponseWriter, r *http.Request, params Li
 	writeResponse(w, http.StatusOK, response)
 }
 
+func (c *Controller) generateResetPasswordToken(email string) (string, error) {
+	secret := c.Auth.SecretStore().SharedSecret()
+	currentTime := time.Now()
+	return GenerateJWTResetPassword(secret, email, currentTime, currentTime.Add(DefaultResetPasswordExpiration))
+}
+
+func (c *Controller) inviteUserRequest(email string) error {
+	// TODO (@shimi9276) create template for sending the email with link for invite using the passed template
+	token, err := c.generateResetPasswordToken(email)
+	if err != nil {
+		return err
+	}
+	err = c.Emailer.SendEmailWithLimit([]string{email}, token, token, nil)
+	if err != nil {
+		return err
+	}
+	c.Logger.WithField("email", email).Info("invite email sent")
+	return nil
+}
+
 func (c *Controller) CreateUser(w http.ResponseWriter, r *http.Request, body CreateUserJSONRequestBody) {
+	invite := swag.BoolValue(body.InviteUser)
+	id := body.Id
+	var email *string
+	if invite {
+		addr, err := mail.ParseAddress(body.Id)
```

Comment:
I do agree Passing addr.Address, (as it is today) should avoid the problem.

Reply:
No, it will create a user with just the email `bg@example.com`, i.e. `Barry Gibbs` is not used at all. We use just the email address. Proper documentation is forthcoming.